Prog/Master School Update Through 2017-18.sas

```diff
@@ -0,0 +1,226 @@
+	if a or c;
+
+	if ui_id_n = 1042000 then do; x = -76.9901 ; y = 38.9024; end;
+	if ui_id_n = 1094702 then do; x = -77.0155929 ; y = 38.9203028; end;
+	if ui_id_n = 1637436 then do; x = -76.9329012 ; y = 38.9065625; end;
+	if ui_id_n = 2108500 then do; x= -77.0283249 ; y = 38.9405892; end;
+	if ui_id_n = 2108600 then do; x = -77.0414395 ; y = 38.8981258; end;
+	if ui_id_n = 2108700 then do; x = -76.9742016 ; y = 38.8556027; end;
+	if ui_id_n = 2108800 then do; x = -76.9986698 ; y = 38.9278646; end;
+
+	year = 2016;
+run;
+
+data schools_17_18;
+	merge schools_17_18_in (in=a) school_locations (in=b) closed_schools (in=c) ;
+	by ui_id_n;
+	if a or c;
+
+	if ui_id_n = 1042000 then do; x = -76.9901 ; y = 38.9024; end;
+	if ui_id_n = 1094702 then do; x = -77.0155929 ; y = 38.9203028; end;
+	if ui_id_n = 1637436 then do; x = -76.9329012 ; y = 38.9065625; end;
+	if ui_id_n = 2108500 then do; x= -77.0283249 ; y = 38.9405892; end;
+	if ui_id_n = 2108600 then do; x = -77.0414395 ; y = 38.8981258; end;
+	if ui_id_n = 2108700 then do; x = -76.9742016 ; y = 38.8556027; end;
+	if ui_id_n = 2108800 then do; x = -76.9986698 ; y = 38.9278646; end;
+	if ui_id_n = 2108701 then do x = -76.9404947; y = 38.8700043; end;
+	if ui_id_n = 2108900 then do; x = -77.0360009 ; y = 38.9260056; end;
+
+	year = 2017;
+run;
+
+
+/* Stack 2015, 2016 and 2017 dataseets */
+data schools_15_18;
+	set schools_15_16 schools_16_17 schools_17_18;
+
+	if SchoolType = 1 then dcps = 1;
+		else if SchoolType = 2 then dcps = 0;
+
+	aud = total;
+
+	ui_id = put(ui_id_n,z7.);
+	drop ui_id_n;
+
+	if master_school_name = " " then do;
+		master_school_name = School_Name;
+	end;
+
+run;
+
+
+/* Use GInside to geocode X/Y to block */
+proc mapimport out=block_shp_in
+  datafile="L:\Libraries\OCTO\Maps\Census_Blocks__2010.shp";  
+run;
+
+proc sort data=block_shp_in out=block_shp (keep = x y geoid10); 
+	by geoid10;
+run;
+
+goptions reset=global border;
+
+proc ginside includeborder
+  data=schools_15_18
+  map=block_shp
+  out=schools_15_18_join;
+  id geoid10;
+run;
+
+
+/* Use DC Data Macros to convert blocks to standard geos */
+data Schools_00_14_geo;
+	set Schools_00_14;
+
+	%Block00_to_anc12;
+	%Block00_to_bpk;
+	%Block00_to_city;
+	%Block00_to_cluster17;
+	%Block00_to_psa12;
+	%Block00_to_stantoncommons;
+	%Block00_to_tr00;
+	%Block00_to_tr10;
+	%Block00_to_vp12;
+	%Block00_to_ward12;
+
+	drop school_name;
+run;
+
+data schools_15_18_geo;
+	set schools_15_18_join;
+
+	geoblk2010 = geoid10;
+
+	%Block10_to_anc12;
+	%Block10_to_bpk;
+	%Block10_to_city;
+	%Block10_to_cluster17;
+	%Block10_to_psa12;
+	%Block10_to_stantoncommons;
+	%Block10_to_tr00;
+	%Block10_to_tr10;
+	%Block10_to_vp12;
+	%Block10_to_ward12;
+
+	keep ui_id master_school_name year dcps aud status
+		 geoblk2010 Anc2012 bridgepk city cluster2017 Psa2012 stantoncommons Geo2000 Geo2010 VoterPre2012 Ward2012;
+run;
+
+
+/* Combine old and new data together */
+data schools_00_18_combined;
+	set Schools_00_14_geo schools_15_18_geo;
```

Comment:
This is not properly creating the entire new MFS because **Schools_00_14_geo** contains only CLOSED school records and, as a result, a lot of observations have been dropped. For example, see *ui_id*=1020700 (BENNING ELEMENTARY SCHOOL), which has dropped data prior to 2008, and *ui_id*=01X02700 (CHILD & FAMILY SERVICES), which is entirely missing from the new file. This needs to be fixed.